qa/tasks/salt_manager.py

```diff
@@ -169,8 +169,26 @@ def gather_logfile(self, logfile):
                 '/home/ubuntu/cephtest/archive/{}'.format(logfile)
                 ])
 
-    def gather_logs(self, logdir):
-        for _remote in self.ctx.cluster.remotes.iterkeys():
+    def gather_logs(self, logdir, archive=None):
+        """
+        Grabs contents of logdir and saves them in /home/ubuntu/cephtest/archive
+        teuthology will harvest them before destroying the remote (target machine).
+
+        logdir can be specified as an absolute path or a relative path. Relative
+        paths are assumed to be under /var/log.
+        """
+        if logdir[:1] == '/':
+            if not archive:
+                raise InternalError((
+                    'Unable to harvest logs from absolute directory ->{}<- '
+                    'because no archive option was passed'
+                    ).format(logdir)
+                    )
+        else:
+            if not archive:
+                archive = logdir
+            logdir = '/var/log/{}'.format(logdir)
+        for _remote in self.ctx.cluster.remotes.keys():
```

Comment:
Adding unrelated code to the cherry-pick is not a good way to resolve the conflict. As I see it, there are two options, here. Either:

* find out which commit added this code in `ses6-downstream-commits` and cherry-pick that commit, first - i.e. make the cherry-pick apply cleanly

or

* make the `.iterkeys` -> `.keys` change manually to resolve the conflict

The first option is higher-risk (it still brings in the unrelated code, which might cause a regression). I would go with the second option.

Reply:
Going with second option.